### ISSUE TITLE
npm moves common dependencies under a dependency on install

### ISSUE THREAD

**Issue by trygveaa** [npm/npm#10258]
When installing a package with npm, it moves all dependencies that are common between your package and the package you are installing to `node_modules` of the package you are installing. This means that you will not have all of your dependencies available in your package anymore.

This bug seems to be introduced in version 3.3.10. It happens on 3.3.10 and 3.3.12, but not on 3.3.9. I'm running node 5.0.0.

To reproduce, create two packages. In directory `a`, create this `package.json`:
```
{
  "name": "a",
  "version": "0.0.1",
  "dependencies": {
    "lodash": "3.10.1"
  }
}
```

In directory `b`, create this `package.json`:
```
{
  "name": "b",
  "version": "0.0.1",
  "dependencies": {
    "a": "../a",
    "lodash": "3.10.1"
  }
}
```

Then run these commands in directory `b`:
```
npm install
npm install ../a
```

After the first command, you will have a `node_modules` containing `a` and `lodash`, which is correct. `node_modules/a` will not contain `lodash`, since that is in the top-level.

After the second command, you will have a `node_modules` containing only `a`. `lodash` has been moved to `node_modules/a/node_modules`, so it is not available from the `b` package anymore.

With npm 3.3.9, `node_modules` is equal after the first and second command (except for some small changes on how `a` is referenced).

I'm not sure, but 971fd47 seems relevant.

**Comment by trygveaa** [comment]
This seems to be a duplicate of #10257, which was just created as well (it didn't exist yet when I checked for similar issues).

**Comment by KenanY** [comment]
@trygveaa Yup, `git bisect` tells me it's https://github.com/npm/npm/commit/971fd47ac9b48cd7aa61d869082c6950639ac366.

**Comment by trygveaa** [comment]
This seems to be fixed by #10338 in v3.4.1. Thanks.
